dig.go

```diff
@@ -664,9 +685,12 @@ func (n *node) Call(c containerStore) error {
 	}
 
 	receiver := newStagingContainerWriter()
-	results := reflect.ValueOf(n.ctor).Call(args)
-	if err := n.resultList.ExtractList(receiver, results); err != nil {
-		return errConstructorFailed{Func: n.location, Reason: err}
+	// Do not call providers if we are in dry mode and  dont want anything to run.
+	if !c.Dry() {
+		results := reflect.ValueOf(n.ctor).Call(args)
+		if err := n.resultList.ExtractList(receiver, results); err != nil {
```

Comment:
So I'm not sure about this, but is it ok to skip this `ExtractList` call? It seems to me that this is where the results from the provide are registered with the container; if you don't register the return types with the container, will an invoke that depends on those types succeed? 

I'm going to investigate in the code as a sanity check, but I feel like the easiest way to confirm would be to add a test case that confirms that in 1) invokes (as well as provides) are not executed and 2) that invokes that depend on return types from provides don't fail in `Dry` mode.

Reply:
Just reviewed your [Fx PR](https://github.com/uber-go/fx/pull/706/), and you've got these test cases transitively covered by those assertions there, so it looks like this won't be a problem. That said, it probably still makes sense to add a parallel test here, or an assertion to `TestDrySucceedsOnPanickingFn` that the invoke also doesn't get run.